test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/EverythingOperationTests.cs

```diff
@@ -0,0 +1,152 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Net;
+using Hl7.Fhir.Model;
+using Microsoft.Health.Fhir.Client;
+using Microsoft.Health.Fhir.Tests.Common.FixtureParameters;
+using Microsoft.Health.Fhir.Tests.E2E.Rest.Search;
+using Microsoft.Health.Test.Utilities;
+using Xunit;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Tests.E2E.Rest
+{
+    [HttpIntegrationFixtureArgumentSets(DataStore.All, Format.Json)]
+    public class EverythingOperationTests : SearchTestsBase<EverythingOperationTestFixture>
+    {
+        public EverythingOperationTests(EverythingOperationTestFixture fixture)
+            : base(fixture)
+        {
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        [HttpIntegrationFixtureArgumentSets(DataStore.CosmosDb, Format.Json)]
+        public async Task GivenAPatientEverythingOperationWithId_WhenSearched_ThenResourcesInScopeShouldBeReturned()
+        {
+            string searchUrl = $"Patient/{Fixture.Patient.Id}/$everything";
+
+            await ExecuteAndValidateBundle(searchUrl, true, 4, Fixture.Patient, Fixture.Organization, Fixture.Device, Fixture.Observation, Fixture.Appointment, Fixture.Encounter);
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        [HttpIntegrationFixtureArgumentSets(DataStore.CosmosDb, Format.Json)]
+        public async Task GivenAPatientEverythingOperationWithNonExistentId_WhenSearched_ThenResourcesInScopeShouldBeReturned()
+        {
+            string searchUrl = $"Patient/{Fixture.NonExistentPatient.Id}/$everything";
+
+            await ExecuteAndValidateBundle(searchUrl, true, 4, Fixture.DeviceOfNonExistentPatient, Fixture.ObservationOfNonExistentPatient);
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        [HttpIntegrationFixtureArgumentSets(DataStore.CosmosDb, Format.Json)]
+        public async Task GivenAEverythingOperationWithUnsupportedType_WhenSearched_ThenBadRequestShouldBeReturned()
+        {
+            string searchUrl = "Observation/bar/$everything";
+
+            using FhirException ex = await Assert.ThrowsAsync<FhirException>(() => Client.SearchAsync(searchUrl));
+
+            Assert.Equal(HttpStatusCode.BadRequest, ex.StatusCode);
```

Comment:
>BadRequest [](http://example.com/codeflow?start=40&length=10)

it should be `not found` since `$everything` defined only for patient resource.

Reply:
As discussed in previous comments, the constraint is now set at route level. If the resource type is not supported, `404 Not Found` is returned.